src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -810,17 +827,54 @@ public struct ReplicationIndexOrTransformerPositions
         private unsafe void ReadDocumentsFromSource(ref UnmanagedWriteBuffer writeBuffer, int replicatedDocs)
         {
             _replicatedDocs.Clear();
+            _replicatedAttachments.Clear();
             for (int x = 0; x < replicatedDocs; x++)
             {
+                var type = *(ReplicationBatchItem.ReplicationItemType*)ReadExactly(sizeof(byte));
+                if (type == ReplicationBatchItem.ReplicationItemType.Attachment)
+                {
+                    var attachment = new ReplicationAttachment
+                    {
+                        TransactionMarker = *(short*)ReadExactly(sizeof(short)),
+                    };
+
+                    var loweredKeySize = *(int*)ReadExactly(sizeof(int));
+                    attachment.LoweredKey = new byte[loweredKeySize];
```

Comment:
We have it here:
https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs#L409